packages/@aws-cdk/custom-resource-handlers/lib/aws-eks/cluster-resource-handler/cluster.ts

```diff
@@ -338,9 +385,38 @@ function analyzeUpdate(oldProps: Partial<EKS.CreateClusterCommandInput>, newProp
     updateVersion: newProps.version !== oldProps.version,
     updateEncryption: JSON.stringify(newEnc) !== JSON.stringify(oldEnc),
     updateLogging: JSON.stringify(newProps.logging) !== JSON.stringify(oldProps.logging),
+    updateTags: JSON.stringify(newProps.tags) !== JSON.stringify(oldProps.tags),
   };
 }
 
 function setsEqual(first: Set<string>, second: Set<string>) {
   return first.size === second.size && [...first].every((e: string) => second.has(e));
 }
+
+function getTagsToUpdate<T extends Record<string, string>>(obj1: T, obj2: T): T {
+  const diff: T = {} as T;
+  // Get all tag keys that are newly added and keys whose value have changed
+  for (const key in obj2) {
+    if (obj2.hasOwnProperty(key)) {
```

Comment:
could you help in understanding what we are trying to validate here since we are iterating over tag key of obj2 or new props as I understand.
also instead of naming it obj1 or obj2 can we rename to something like oldTags and newTags since that's the only type of object we are expecting currently ?

Reply:
It is to check if there are new tags (tags with new keys) which are not present in the old props. 
For update, at this part, getting both the tags that are updated and newly added. 

```
old - 
{
  foo: "bar"
}
new - 
{
 foo: "bars"
 hello: "world"
}
```
with the above props, we have to send both the tag keys [foo, hello] in the tagresource API

I will change the names to oldTags and newTags

Reply:
Thank you, changing obj names would be helpful

Reply:
The names have been changed, along with more comments